### PR TITLE
ZEPPELIN-546 Enables interpreter library loading from maven repository

### PR DESCRIPTION
**Pull request by minahlee**
### What is this PR for?
This PR enables library loading from maven repository to load interpreter binaries.
To leverage current spark interpreter's library loading, moved org.apache.zeppelin.spark.dep(under spark) to org.apache.zeppelin.dep(under zeppelin-interpreter).
Making REST API and loading interpreter property on runtime will be the next step to complete [ZEPPELIN-546](https://issues.apache.org/jira/browse/ZEPPELIN-546?jql=project%20%3D%20ZEPPELIN) and will be handled in different PR.

### What type of PR is it?
Feature

### Is there a relevant Jira issue?
[ZEPPELIN-546](https://issues.apache.org/jira/browse/ZEPPELIN-546?jql=project%20%3D%20ZEPPELIN)

### Questions:
* Does the licenses files need update? No (dependencies added to `zeppelin-interpreter/pom.xml` is already used in other module)
* Is there breaking changes for older versions? No
* Does this needs documentation? Yes (will be addressed in another PR)

### CODE REVIEW

**Comment by bzz** [comment]
Looks great! A quick question if it's ready to merge: `zeppelin-interpreter/pom.xml` was changed with dependencies added but you think there is no need to mention any of those in licence files?

Also, is there a reason not to handle documentation update in the same PR as feature and it's tests?

**Comment by minahlee** [comment]
@bzz Thanks for review. Dependencies that added to `zeppelin-interpreter/pom.xml` is already used in `spark/pom.xml`, so I answered 'no' for license update question. I update PR description for the ones who might be confused like you. Basic test is already included in this PR. About the docs @AhyoungRyu will contribute it but if she commits on this PR, her contribution won't be counted. That's the only reason we separate the PR for docs.

**Comment by AhyoungRyu** [comment]
@bzz  Yes, @minahlee is right. I'm working on it : )
 I'll commit PR for docs soon, after this PR merged.

**Comment by corneadoug** [comment]
@minahlee still doesn't make much sense to separate the docs into a different PR.
At least create both PR so that we can merge both  at the same time.

**Comment by Leemoonsoo** [comment]
I think how to split large work into small peices of PRs is up to contributors. (although smaller PR helps review process)
If necessary documentation for this PR comes with different PR, we can wait for that and merge both PR together.

**Comment by corneadoug** [comment]
@Leemoonsoo Yes, it will be better to have both at the same time to merge them together.
Its just to keep features documented as we include them in the code, that way people know about it and also know how to use it. It can also help a lot for the reviewer to test the feature. (Its not really explained right now)

**Comment by bzz** [comment]
@minahlee @AhyoungRyu - thanks for explanations, makes perfect sense to have 2 PRs for 2 ppl!

Looks good to me.

**Comment by jongyoul** [comment]
LGTM!! It also makes any kind of jars loaded easily. I could load JDBC libraries using this feature.

**Comment by minahlee** [comment]
Rebased master branch and CI build passed

**Comment by Leemoonsoo** [comment]
Great. LGTM

File: zeppelin-interpreter/src/main/java/org/apache/zeppelin/dep/DependencyResolver.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+/**
+ * Deps resolver.
+ * Add new dependencies from mvn repo (at runtime) to Zeppelin.
+ */
+public class DependencyResolver extends AbstractDependencyResolver {
+  Logger logger = LoggerFactory.getLogger(DependencyResolver.class);
+
+  private final String[] exclusions = new String[] {"org.apache.zeppelin:zeppelin-zengine",
+                                                    "org.apache.zeppelin:zeppelin-interpreter",
```
Comment (pooja155) [inline zeppelin-interpreter/src/main/java/org/apache/zeppelin/dep/DependencyResolver.java]: hi, was there any reason to not include all the exclusions mentioned in SparkDependencyResolver.java here?
"org.scala-lang:scala-library","org.scala-lang:scala-compiler","org.scala-lang:scala-reflect","org.scala-lang:scalap" - these are present in the exclusions in SparkDependencyResolver.java but not here.

Comment (jongyoul) [reply zeppelin-interpreter/src/main/java/org/apache/zeppelin/dep/DependencyResolver.java]: AFAIK, spark already has scala related packages but this doesn't have them.

**Pull request closed without merging**